### PR TITLE
add walkthrough of toml

### PR DESCRIPTION
**Pull request by leoebfolsom**
I've added more detail to the docs explaining how to use a TOML file. In the future, I'd like to create a Loom, but I think we could consider merging this as an incremental improvement. I'm open to feedback and ideas.

Medium-term, I'd like to restructure this so the TOML config setup and walkthrough is not buried far down the `How to Use` page. However, I'd like to prioritize this update and then separately consider the structure, since I believe users can benefit today from this additional detail about how to use a config file.

### CODE REVIEW

**Review by williebsweet — APPROVED** [review]
Made one slight change.

File: docs/os_diff/how_to_use.md [hunk 1]
```diff
@@ -126,54 +126,103 @@ it's recommended to surround them with quotes. Alternatively, you may [provide t
...
-Finally, CLI switches have the final say, and will override the settings defined by the configuration file, and the current run.
+By setting up a TOML file like this:
 
-Example TOML file:
+```
+# DATABASE CONNECTION INFORMATION
+# In this section, you can specify one or more databases that your runs (defined later in the file) can connect to.
+
+# Specify the database connection information
+[database.db_postgres]
```
Comment (erezsh) [inline docs/os_diff/how_to_use.md:157]: Not sure about `db_postgres` as a good naming convention, we already know it's a database.

File: docs/os_diff/how_to_use.md [hunk 2]
```diff
@@ -126,54 +126,103 @@ it's recommended to surround them with quotes. Alternatively, you may [provide t
...
 
-```toml
-# Specify the connection params to the test database.
-[database.test_postgresql]
-driver = "postgresql"
-user = "postgres"
-password = "Password1"
+  driver = "postgresql"
+  database = "diff_test"
+  user = "your_username"
```
Comment (erezsh) [inline docs/os_diff/how_to_use.md:161]: It's a little funny you use a generic username but specific password :)

File: docs/os_diff/how_to_use.md [hunk 3]
```diff
@@ -126,54 +126,103 @@ it's recommended to surround them with quotes. Alternatively, you may [provide t
...
-verbose = true
-
-# Specify params for a run 'test_diff'.
-[run.test_diff]
-verbose = false
-# Source 1 ("left")
-1.database = "test_postgresql"                      # Use options from database.test_postgresql
-1.table = "rating"
-# Source 2 ("right")
-2.database = "postgresql://postgres:Password1@/"    # Use URI like in the CLI
```
Comment (erezsh) [inline docs/os_diff/how_to_use.md:168]: You removed the information that the `database` keyword accepts a URL

If not here, it should appear somehwere, I think.

**Review by erezsh — COMMENTED** [review]
I like it!

I only have a few comments.

Also, I think it could benefit from a couple more substantial examples, which include non-standard parameters, to give a feel of why it's useful.

Especially if afterwards you show the equivalent URL 😆

**Comment by erezsh** [comment]
Oh, also that's a bit premature, no?

```md
In both cases, we recommend storing sensitive information such as passwords as [environmental variables](https://linuxhint.com/set-environment-variable-zsh/) in a file such as .zshrc.
```

File: docs/os_diff/how_to_use.md (hunk 1 above)
Comment (leoebfolsom) [reply docs/os_diff/how_to_use.md:157]: I can update that 👍

File: docs/os_diff/how_to_use.md (hunk 2 above)
Comment (leoebfolsom) [reply docs/os_diff/how_to_use.md:161]: yeah yeah, I'll update that too :)

File: docs/os_diff/how_to_use.md (hunk 3 above)
Comment (leoebfolsom) [reply docs/os_diff/how_to_use.md:168]: I'll make a note of it 👍

**Comment by leoebfolsom** [comment]
> Oh, also that's a bit premature, no?
> 
> ```gfm
> In both cases, we recommend storing sensitive information such as passwords as [environmental variables](https://linuxhint.com/set-environment-variable-zsh/) in a file such as .zshrc.
> ```

@erezsh I actually removed that line in a subsequent commit after our discussion 👍

**Comment by leoebfolsom** [comment]
> Also, I think it could benefit from a couple more substantial examples, which include non-standard parameters, to give a feel of why it's useful. 

@erezsh could you please give me an example of what you mean by non-standard parameters? Ideally, something that you think might resonate with users, leading to a light-bulb moment of "ah, _this_ is when a TOML file would be really handy." 💡

**Comment by erezsh** [comment]
@leoebfolsom  Here are two examples of providing auth info in config, one of presto, and one of snowflake -

```toml
[database.presto]
driver = "presto"
host = "<presto hostname>"
port = "<presto port>"
user = "<presto user>"
password = "<presto password>"
catalog = "<presto catalog>"
schema = "<presto schema>"
cert = "<path/to/presto/certificate.cer>"
auth = "basic"
http_scheme = "https"
source = "odbc"
```

```toml
[database.snowflake]
driver = "snowflake"
account = "myacc"
database = "TESTS"
user = "erez"
warehouse = "mywh"
role = "myrole"
schema = "PUBLIC"
key = "snowflake_rsa_key.p8"
```

**Pull request merged**